GeofenceDisplay/src/geofence-display.ts

```diff
@@ -1,155 +1,111 @@
-    // }
-
-    // self.clearGeofences = function () {
-    //   for (let geofenceID in self.geofence.map) {
-    //     let geofence = self.geofence.map[geofenceID]
-    //     delete self.geofence.map[geofenceID]
-    //     if (geofence && geofence.hide) {
-    //       geofence.hide()
-    //     }
-    //   }
-    // }
-
-    // // TODO: add geofencePane to div
-
-    return _map
+    let self = this
+    self.options.debug && console.log('GeofenceDisplay onAdd function called.')
+    self.options.debug &&
+      console.log(
+        'Local geofence variable before add:',
+        self._state.geofenceByID
+      )
+    self.options.geofences.forEach((fence: any) => {
+      let geo = self.options.geofences[fence]
+      let geofence = new Geofence(geo, { map: _map })
+      self._state.geofenceByID[geo.id] = geofence
+      geofence.show()
+    })
+    self.options.debug &&
+      console.log(
+        'Local geofence variable after add:',
+        self._state.geofenceByID
+      )
   },
 
   onRemove: function (this: any, _map: any) {
-    console.log('PGeoD onRemove')
-  },
-
-  clear: function (this: any) {
-    console.log('PGeoD clear')
+    let self = this
+    self.options.debug &&
+      console.log('GeofenceDisplay onRemove function called.')
+    self.options.debug &&
+      console.log(
+        'Local geofence variable before remove:',
+        self._state.geofenceByID
+      )
+    for (let id in self._state.geofenceByID) {
+      self._state.geofenceByID[id].hide()
+      delete self._state.geofenceByID[id]
+    }
+    self.options.debug &&
+      console.log(
+        'Local geofence variable after remove:',
+        self._state.geofenceByID
+      )
   },
 })
 
 class Geofence {
-  ent = null
-  ctx = null
-  poly = L.polygon(
-    [
-      [37, -109.05],
-      [41, -109.03],
-      [41, -102.05],
-      [37, -102.04],
-    ],
-    { color: 'red' }
-  )
+  ent: any = null
+  ctx: any = null
+  poly: any = null
 
   constructor(ent: any, ctx: any) {
     this.ent = ent
     this.ctx = ctx
   }
 
-  show(layer: any) {
-    if (null == this.poly) {
-      // let polyCoords = convertPoly(this.ctx.cfg.mapImg, this.ent.polygon)
-      // this.poly = L.polygon(polyCoords, {
-      //   pane: 'geofence',
-      //   color: this.ctx.cfg.geofence.color,
-      // })
-      // if (this.ctx.cfg.geofence.click.active) {
-      //   this.poly.on('click', this.onClick.bind(this))
-      // }
-      // let tooltip = L.tooltip({
-      //   pane: 'geofenceLabel',
-      //   permanent: true,
-      //   direction: 'center',
-      //   opacity: 1,
-      //   className: 'polygon-labels',
-      // })
-      // this.poly.bindTooltip(tooltip)
-      // tooltip.setContent(
-      //   '<div class="' +
-      //     'leaflet-zoom-animated ' +
-      //     'plantquest-geofence-label ' +
-      //     `">${this.ent.title}</div>`
-      // )
+  show() {
+    let self = this
+    if (null == self.poly) {
+      let polyCoords = self.ent.latlngs
```

Comment:
this setup code should go into the constructor
show only needs the addTo(map)

Reply:
Ah - referenced in this comment: https://github.com/plantquest/leaflet-plugins/pull/8/files#r1392422291